### PR TITLE
feature/WAL-320 

### PR DESCRIPTION
**Pull request by BtencateSphereon**

### CODE REVIEW

File: src/localization/translations/en.json [hunk 1]
```diff
@@ -134,5 +134,10 @@
     "onboard_summary_title": "Your entered details",
     "onboard_summary_subtitle": "When you press on finalize, you accept the data you entered and you will proceed to the login screen. The personal data can be changed in the settings.",
     "onboard_summary_info_tab_header_label": "Your information",
-    "onboard_summary_button_caption": "Finalize and go to my wallet"
+    "onboard_summary_button_caption": "Finalize and go to my wallet",
+    "credentials_required_title": "Required credentials",
+    "credentials_required_subtitle": "The verifying party: {0} would like to receive the following information from you for verification.",
```
Comment (nklomp) [inline src/localization/translations/en.json:139]: `The verifying party:` can be removed in its entirety

File: src/localization/translations/nl.json [hunk 2]
```diff
@@ -134,5 +134,10 @@
     "onboard_summary_title": "Uw ingevulde gegevens",
     "onboard_summary_subtitle": "Als je op voltooien drukt, accepteer je de gegevens die je hebt ingevoerd en ga je verder naar het inlogscherm. De persoonlijke gegevens kunnen worden gewijzigd in de instellingen.",
     "onboard_summary_info_tab_header_label": "Uw gegevens",
-    "onboard_summary_button_caption": "Voltooi en ga naar mijn wallet"
+    "onboard_summary_button_caption": "Voltooi en ga naar mijn wallet",
+    "credentials_required_title": "Vereiste credentials",
+    "credentials_required_subtitle": "De verifiërende partij: {0} wil graag de volgende informatie van u ontvangen ter verificatie.",
```
Comment (nklomp) [inline src/localization/translations/nl.json:139]: `De verifiërende partij:` can be removed in its entirety

File: src/screens/SSICredentialsRequiredScreen/index.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  const [selectedCredentials, setSelectedCredentials] = useState(new Map<string, Array<VerifiableCredential>>())
+  const [availableCredentials, setAvailableCredentials] = useState(new Map<string, Array<VerifiableCredential>>())
+  const pex = new PEX()
+
+  useEffect(() => {
+    getVerifiableCredentialsFromStorage()
+      .then((vcs: Array<UniqueVerifiableCredential>) => {
+        const originalVcs: Array<OriginalVerifiableCredential> = vcs.map((vc: UniqueVerifiableCredential) => vc.verifiableCredential as OriginalVerifiableCredential)
+        const credentials = new Map<string, Array<VerifiableCredential>>()
+        presentationDefinition.input_descriptors.forEach((inputDescriptor: InputDescriptorV1) => {
```
Comment (nklomp) [inline src/screens/SSICredentialsRequiredScreen/index.tsx:41]: Hmzz, given the only direct difference between v1 and v2 input descriptors is the potential presence of the schema property (v1 only), IMO we should immediately accomodate for both at least in this screen. A simple union should suffice for now
Another difference is down the line in the constraints, but that is not something this screen is concerned about

File: src/screens/SSICredentialsRequiredScreen/index.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    props.navigation.goBack()
+  }
+
+  // TODO add support for multiple versions of pex, currently using V1
+  const renderItem = (itemInfo: ListRenderItemInfo<InputDescriptorV1>): JSX.Element => {
+    return (
+      <SSICredentialRequiredViewItem
+        id={itemInfo.item.id}
+        title={itemInfo.item.name || itemInfo.item.id}
+        available={availableCredentials.get(itemInfo.item.id) ? availableCredentials.get(itemInfo.item.id)! : []}
```
Comment (nklomp) [inline src/screens/SSICredentialsRequiredScreen/index.tsx:72]: .has() for the boolean expression

File: src/screens/SSICredentialsRequiredScreen/index.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            : []
+          credentials.set(inputDescriptor.id, matchedCredentials)
+        })
+        setAvailableCredentials(credentials);
+      })
+  }, []);
+
+  useEffect(() => {
+    const credentials = new Map<string, Array<VerifiableCredential>>()
+    presentationDefinition.input_descriptors.forEach((inputDescriptor: InputDescriptorV1) => credentials.set(inputDescriptor.id, []))
```
Comment (nklomp) [inline src/screens/SSICredentialsRequiredScreen/index.tsx:58]: Hmzz, given the only direct difference between v1 and v2 input descriptors is the potential presence of the schema property (v1 only), IMO we should immediately accomodate for both at least in this screen. A simple union should suffice for now
Another difference is down the line in the constraints, but that is not something this screen is concerned about

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks. I think for this screen supporting both V1 and V2 is not really much additional work, given the direct object property difference is only the schema property. Hence this screen should be able to accommodate v1 and v2 already.

File: src/localization/translations/en.json (hunk 1 above)
Comment (BtencateSphereon) [reply src/localization/translations/en.json]: removed The verifying party:

File: src/localization/translations/nl.json (hunk 2 above)
Comment (BtencateSphereon) [reply src/localization/translations/nl.json]: removed The verifying party:

File: src/screens/SSICredentialsRequiredScreen/index.tsx (hunk 3 above)
Comment (BtencateSphereon) [reply src/screens/SSICredentialsRequiredScreen/index.tsx]: added v2 as well

File: src/screens/SSICredentialsRequiredScreen/index.tsx (hunk 4 above)
Comment (BtencateSphereon) [reply src/screens/SSICredentialsRequiredScreen/index.tsx]: updated

File: src/screens/SSICredentialsRequiredScreen/index.tsx (hunk 5 above)
Comment (BtencateSphereon) [reply src/screens/SSICredentialsRequiredScreen/index.tsx]: added v2 as well

**Review by nklomp — APPROVED** [review]
LGTM

**Pull request merged**